flake.nix

```diff
@@ -0,0 +1,448 @@
+        };
+        pkgs-bitcoind = import nixpkgs-bitcoind {
+          inherit system overlays;
+        };
+        pkgs-kitman = import nixpkgs-kitman {
+          inherit system;
+        };
+
+        # Signed Commits
+        signed-commits = pkgs.writeShellApplication {
+          name = "signed-commits";
+          runtimeInputs = [ pkgs.git ];
+          text = ''
+            # Function to verify the signature of the last commit
+            verify_signature() {
+            	local commit_hash
+              commit_hash=$(git rev-parse HEAD)
+            	if ! git verify-commit "$commit_hash"; then
+            		echo "Error: Last commit ($commit_hash) is not signed."
+            		exit 1
+            	fi
+            }
+
+            # Verify the signature of the last commit
+            verify_signature
+
+            # Allow the push to proceed if the signature is valid
+            exit 0
+          '';
+        };
+
+        # Toolchains
+        # latest stable
+        rustTarget = pkgs.rust-bin.stable.latest.default;
+        # we pin clippy instead of using "stable" so that our CI doesn't break
+        # at each new cargo release
+        rustClippyTarget = pkgs.rust-bin.stable."1.67.0".default;
+        # MSRV
+        rustMSRVTarget = pkgs.rust-bin.stable."1.57.0".default;
+        # WASM
+        rustWASMTarget = pkgs.rust-bin.stable.latest.default.override {
+          targets = [ "wasm32-unknown-unknown" ];
+        };
+        # LLVM code coverage
+        rustLLVMTarget = pkgs.rust-bin.stable.latest.default.override {
+          extensions = [ "llvm-tools" ];
+        };
+        # Nighly Docs
+        rustNightlyTarget = pkgs.rust-bin.selectLatestNightlyWith (toolchain: toolchain.default);
+
+        # Rust configs
+        craneLib = (crane.mkLib pkgs).overrideToolchain rustTarget;
+        # Clippy specific configs
+        craneClippyLib = (crane.mkLib pkgs).overrideToolchain rustClippyTarget;
+        # MSRV specific configs
+        # WASM specific configs
+        # craneUtils needs to be built using Rust latest (not MSRV/WASM)
+        # check https://github.com/ipetkov/crane/issues/422
+        craneMSRVLib = ((crane.mkLib pkgs).overrideToolchain rustMSRVTarget).overrideScope' (final: prev: { inherit (craneLib) craneUtils; });
+        craneWASMLib = ((crane.mkLib pkgs).overrideToolchain rustWASMTarget).overrideScope' (final: prev: { inherit (craneLib) craneUtils; });
+        # LLVM code coverage
+        craneLLVMLib = (crane.mkLib pkgs).overrideToolchain rustLLVMTarget;
+        craneNightlyLib = (crane.mkLib pkgs).overrideToolchain rustNightlyTarget;
+
+        # Common inputs for all derivations
+        buildInputs = [
+          # Add additional build inputs here
+          pkgs-bitcoind.bitcoind
+          pkgs-kitman.esplora
+          pkgs.openssl
+          pkgs.openssl.dev
+          pkgs.pkg-config
+          pkgs.curl
+          pkgs.libiconv
+        ] ++ lib.optionals isDarwin libsDarwin;
```

Comment:
**suggestion:** I think I would just inline the check int `libsDarwin` (so it's empty when not on Darwin).

Reply:
I don't get. How I would do that? And why that check is important?

Reply:
```nix
let libsDarwin = lib.optionals isDarwin [ ... ];
```

so that `libsDarwin` contain darwin specific libs on Darwin, and is empty otherwise.

Reply:
Like in 040a8e37a364920b6647fbc6c676cb6c98d291eb?

Reply:
:+1: